src/app/WriteClient.h

```diff
@@ -135,6 +135,74 @@ class WriteClient : public Messaging::ExchangeDelegate
         return CHIP_NO_ERROR;
     }
 
+    /**
+     *  Encode an attribute value that can be directly encoded using TLVWriter::Put
+     */
+    template <class T>
+    CHIP_ERROR EncodeAttributeWritePayload(const chip::app::AttributePathParams & attributePath, const T & value)
+    {
+        ReturnErrorOnFailure(PrepareNewChunk());
+        return EncodeSingleAttributeDataIB(
+            ConcreteDataAttributePath(attributePath.mEndpointId, attributePath.mClusterId, attributePath.mAttributeId), value);
+    }
+
+    /**
+     *  Encode an attribute value that can be directly encoded using TLVWriter::Put
+     *
+     *  Note: When encoding lists with this function, you may receive more than one write state for each list entry. You can refer
+     * to ChunkedWriteCallback.h for a high level API which will merge status code for chunked write requests.
+     */
+    template <class T>
+    CHIP_ERROR EncodeAttributeWritePayload(const chip::app::AttributePathParams & attributePath, const DataModel::List<T> & value)
+    {
+        TLV::TLVWriter backupWriter;
+        ConcreteDataAttributePath path =
+            ConcreteDataAttributePath(attributePath.mEndpointId, attributePath.mClusterId, attributePath.mAttributeId);
+
+        ReturnErrorOnFailure(PrepareNewChunk());
```

Comment:
Why are we starting a new message every time we write a new attribute? Why can't we stuff multiple attributes in a single message?

Reply:
Previously for convenient, removed this with a few helper functions.